### PR TITLE
netlist: standalone makefile and prototype modifications

### PR DESCRIPTION
**Pull request by couriersud**
- update netlist makefile to more recent compilers
- prototypes for NETLISTs in macros are now created create_devinc.py
- apply ctidy recommendations

### CODE REVIEW

File: src/lib/netlist/build/create_devinc.py [hunk 1]
```diff
@@ -72,6 +79,11 @@ def process_file(srcfile):
                         m = re.match(r"(static)*TRUTHTABLE_START\((\w+),(\w+),(\w+),([a-zA-Z0-9_+@,]*)", ls)
                         if m != None:
                             process_entry(srcfile, m.group(2), m.group(5))
+                        else:
+                            #m = re.match(r"EXTERNAL_SOURCE\((\w+)\)", ls)
+                            m = re.match(r"NETLIST_START\((\w+)\)", ls)
+                            if m != None:
```
Comment (cuavas) [inline src/lib/netlist/build/create_devinc.py:85]: You probably want `m is not None` here – equality comparison with `None` can have weird semantics, you generally want to check whether the thing is or isn’t the special `None` object.

File: src/lib/netlist/core/base_objects.h [hunk 2]
```diff
@@ -21,10 +21,7 @@
 
 #include <unordered_map>
 
-namespace netlist
-{
-
-	namespace detail {
+namespace netlist::detail {
 
 		template <typename C, typename T>
```
Comment (cuavas) [inline src/lib/netlist/core/base_objects.h:26]: This has an extra level of indent now that you’re using a nested namespace definition.

File: src/lib/netlist/core/devices.h [hunk 3]
```diff
@@ -21,10 +21,8 @@
 // Namespace starts
 //============================================================
 
-namespace netlist
+namespace netlist::devices
 {
-	namespace devices
-	{
 		// -----------------------------------------------------------------------------
```
Comment (cuavas) [inline src/lib/netlist/core/devices.h:26]: This header also has the extra level of indent with the nested namespace definition.

File: src/lib/netlist/core/queue.h [hunk 4]
```diff
@@ -23,9 +23,8 @@
 #include <vector>
 #include <queue>
 
-namespace netlist
+namespace netlist::detail
 {
-	namespace detail {
 		// Use timed_queue_heap to use stdc++ heap functions instead of linear processing.
```
Comment (cuavas) [inline src/lib/netlist/core/queue.h:28]: This file ends up with an extra level of indentation, too.

File: src/lib/netlist/devices/nld_9316_base.hxx [hunk 5]
```diff
@@ -8,10 +8,8 @@
 #include "nl_base.h"
 #include "nl_factory.h"
 
-namespace netlist
+namespace netlist::devices
 {
-	namespace devices
-	{
 
 		template <unsigned N>
```
Comment (cuavas) [inline src/lib/netlist/devices/nld_9316_base.hxx:14]: This is another file with an extra level of indentation now.

File: src/lib/netlist/nl_factory.cpp [hunk 6]
```diff
@@ -11,8 +11,7 @@
 #include "nl_setup.h"
 #include "plib/putil.h"
 
-namespace netlist {
-namespace factory {
+namespace netlist::factory {
 
 	// FIXME: this doesn't do anything, check how to remove
```
Comment (cuavas) [inline src/lib/netlist/nl_factory.cpp:16]: This is another class that’s got an extra level of indentation now.

Comment (cuavas) [reply src/lib/netlist/nl_factory.cpp:16]: Oh sorry, this one actually isn’t – it’s just GitHub displaying different files with different tab widths.

File: src/lib/netlist/build/create_devinc.py (hunk 1 above)
Comment (couriersud) [reply src/lib/netlist/build/create_devinc.py:85]: Applied. Thanks.

File: src/lib/netlist/core/base_objects.h (hunk 2 above)
Comment (couriersud) [reply src/lib/netlist/core/base_objects.h:26]: I wanted to fix indentation in a separate step. Changed now.

File: src/lib/netlist/core/devices.h (hunk 3 above)
Comment (couriersud) [reply src/lib/netlist/core/devices.h:26]: applied

File: src/lib/netlist/devices/nld_9316_base.hxx (hunk 5 above)
Comment (couriersud) [reply src/lib/netlist/devices/nld_9316_base.hxx:14]: Applied

**Comment by couriersud** [comment]
All review comments were applied. Checks passing.

**Pull request merged**